dev_environment/setup_dev_env.sh

```diff
@@ -0,0 +1,178 @@
+else
+  git clone https://github.com/l3af-project/l3afd.git
+  git clone https://github.com/l3af-project/l3af-arch.git
+fi
+
+# Copy grafana configs and dashboards into place
+if [ -d "/var/lib/grafana/dashboards" ];
+then
+  echo "grafana directory already existed"
+else
+  mkdir -p /var/lib/grafana/dashboards
+fi
+chown grafana:grafana /var/lib/grafana/dashboards
+cp /root/l3af-arch/dev_environment/cfg/grafana/dashboards/*.json /var/lib/grafana/dashboards
+chown grafana:grafana /var/lib/grafana/dashboards/*.json
+cp /root/l3af-arch/dev_environment/cfg/grafana/provisioning/dashboards/l3af.yaml /etc/grafana/provisioning/dashboards
+chown root:grafana /etc/grafana/provisioning/dashboards/*.yaml
+cp /root/l3af-arch/dev_environment/cfg/grafana/provisioning/datasources/l3af.yaml /etc/grafana/provisioning/datasources
+chown root:grafana /etc/grafana/provisioning/datasources/*.yaml
+
+# Copy prometheus config and restart prometheus
+cp /root/l3af-arch/dev_environment/cfg/prometheus.yml /etc/prometheus/prometheus.yml
+if uname -a | grep -q 'WSL';
+then
+  echo "WSL DETECTED"
+  apt-get install daemon
+  #start/restart prometheus-node-exporter
+  /etc/init.d/prometheus-node-exporter start
+  /etc/init.d/prometheus-node-exporter stop
+  /etc/init.d/prometheus-node-exporter start
+
+  # Start and enable Grafana
+  sleep 1
+  /etc/init.d/grafana-server stop || true
+  /etc/init.d/grafana-server stop || true
+  /etc/init.d/grafana-server start || true
+else
+  systemctl daemon-reload
+  systemctl restart prometheus prometheus-node-exporter
+
+  # Start and enable Grafana
+  systemctl start grafana-server
+  systemctl enable grafana-server.service
+fi
+
+# Get Linux source code to build our eBPF programs against
+if [ ! -d "/usr/src/linux" ];
+then
+  git clone --branch v5.1 --depth 1 https://github.com/torvalds/linux.git /usr/src/linux
+fi
+LINUX_SRC_DIR=/usr/src/linux
+cd $LINUX_SRC_DIR
+make defconfig
+
+if [ ! -d "/var/log/tb/l3af" ];
+then
+  mkdir -p /var/log/tb/l3af
+fi
+if [ ! -d "/var/l3afd" ];
+then
+  mkdir -p /var/l3afd
+fi
+
+BUILD_DIR=$LINUX_SRC_DIR/samples/bpf/
+
+# Where to store the tar.gz build artifacts
+BUILD_ARTIFACT_DIR=/srv/l3afd
+if [ ! -d "$BUILD_ARTIFACT_DIR" ];
+then
+  mkdir -p $BUILD_ARTIFACT_DIR
+fi
+
+cd $BUILD_DIR
+
+# Get the eBPF-Package-Repository repo containing the eBPF programs
+if [ ! -d "$BUILD_DIR/eBPF-Package-Repository" ];
+then
+  git clone https://github.com/l3af-project/eBPF-Package-Repository.git
+fi
+cd eBPF-Package-Repository
+
+# declare an array variable
+declare -a progs=("xdp-root" "ratelimiting" "connection-limit" "tc-root" "ipfix-flow-exporter")
+
+# now loop through the above array and build the L3AF eBPF programs
+for prog in "${progs[@]}"
+do
+	cd $prog
+	make
+	PROG_ARTIFACT_DIR=$BUILD_ARTIFACT_DIR/$prog/latest/focal
+	mkdir -p $PROG_ARTIFACT_DIR
+	mv *.tar.gz $PROG_ARTIFACT_DIR
+	cd ../
+done
+
+# Compile L3AFD daemon and start the control plan
+cd /root/l3afd
+make install
+cd ../go/bin/
```

Comment:
Can this get l3afd from a build/release artifact rather than having to compile it to run the script?
Ok if that's done in a separate PR though

Reply:
This will be done in a separate PR if necessary.  On the other hand, for setting up development environment script, I would prefer to building the code instead of downloading from the release artifact.  It is a good test whether the environment got set up correctly or not.